src/zmalloc.c

```diff
@@ -188,16 +197,24 @@ void zfree_no_tcache(void *ptr) {
  * '*usable' is set to the usable size if non NULL. */
 static inline void *ztrycalloc_usable_internal(size_t size, size_t *usable) {
     /* Possible overflow, return NULL, so that the caller can panic or handle a failed allocation. */
-    if (size >= SIZE_MAX/2) return NULL;
+    if (size >= SIZE_MAX/2) {
+        if (usable) *usable = 0;
```

Comment:
do it once at the beginning?

Reply:
I see that other functions are like this, so i going this way.

Reply:
i'm not certain we need to bother setting `usable` when we return NULL.
while it's normally a good idea to always avoid uninitialized variables, i don't think in this case we need to bother.
do you have any reason to think this is needed?

Reply:
i just thought it could save one line.

Reply:
i guess oran is actually talking about me.


there is no particular reason. just because the other function like ztryrealloc_usable_internal is doing the same way. i feels it is the same cleanup and the code remains consitent.
```
/* Try reallocating memory, and return NULL if failed.
 * '*usable' is set to the usable size if non NULL. */
static inline void *ztryrealloc_usable_internal(void *ptr, size_t size, size_t *usable) {

    /* Possible overflow, return NULL, so that the caller can panic or handle a failed allocation. */
    if (size >= SIZE_MAX/2) {
        zfree(ptr);
        if (usable) *usable = 0;
        return NULL;
    }
```

Reply:
well, i think `realloc` is a bit different because it can also act as either `malloc` (new allocation) or `free`.
i.e. given a NULL pointer as input it acts as `malloc`, and given a non-NULL and 0 size, it acts as `free.
so in that sense, a NULL return value from `realloc` is not a necessarily failure.

i'm not sure if that's a good justification, since our code does that on failures too (not just when the requested size is 0), and also unlike `free` the returned size in `realloc` is the size of the new allocation, not the released one.

but anyway, i feel that we can reduce the diff here, and keep realloc different from the other ones in that sense.